hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/compact/CompactFunction.java

```diff
@@ -133,4 +133,14 @@ private HoodieWriteConfig reloadWriteConfig() throws Exception {
   public void setExecutor(NonThrownExecutor executor) {
     this.executor = executor;
   }
+
+  @Override
+  public void close() throws Exception {
+    if (this.asyncCompaction) {
+      this.executor.close();
```

Comment:
`this.asyncCompaction` -> `this.executor != null`

Reply:
And we can fix the `ClusteringOperator#close` as well.

Reply:
Got it, good idea

Reply:
Added